apps/hash-frontend/src/pages/notes.page.tsx

```diff
@@ -99,48 +106,6 @@ const NotesPage: NextPageWithLayout = () => {
           ...zeroedGraphResolveDepths,
           isOfType: { outgoing: 1 },
         },
-        temporalAxes: currentTimeInstantTemporalAxes,
-      },
-    },
-    fetchPolicy: "cache-and-network",
-  });
-
-  const quickNotesSubgraph = quickNotesData?.structuralQueryEntities
-    .subgraph as Subgraph<EntityRootType> | undefined;
-
-  const quickNoteEntities = useMemo(
-    () => (quickNotesSubgraph ? getRoots(quickNotesSubgraph) : undefined),
-    [quickNotesSubgraph],
-  );
-
-  const [
-    previouslyFetchedQuickNotesAllVersionsData,
-    setPreviouslyFetchedQuickNotesAllVersionsData,
-  ] = useState<StructuralQueryEntitiesQuery>();
-
-  const { data: quickNotesAllVersionsData } = useQuery<
-    StructuralQueryEntitiesQuery,
-    StructuralQueryEntitiesQueryVariables
-  >(structuralQueryEntitiesQuery, {
```

Comment:
Apologies Ciaran, whilst testing I realised what the purpose of having this in two queries was: to first determine which quick note entities are archived, then fetch all revisions of those entities. The reason we can't combine these into a single query is because archived quick notes will always be included in the result, as they have prior revisions where archived property will be `undefined` or `false`. This results in archiving quick notes to appear broken in this PR.

We could maybe get around this by checking if the latest revision of the quick note is included in the subgraph, but this would mean we are over-fetching the number of quick notes on every page load. Unless we can somehow specify in the filter that the latest revision of the entity cannot be archived, I think it's probably best to go back to using two queries.

Reply:
Ahh, that makes sense. I think we're still better off using one query – although we'll be fetching quick notes that are archived, I expect those will be a minority of all a user's quick notes, and it will be slower to load to fetch the non-archived quick notes and _then_ fetch all the revisions of those, rather than just fetching all the revisions of all quick notes. A quick note can only have more than one revision if it is archived, so they are all likely going to have 1 or 2 revisions to fetch. I think including archived in one request will be quicker than a waterfall of 2 requests.

I've done this in these commits https://github.com/hashintel/hash/pull/3525/files/8bb1feafcf87a69655b856d7368470499fd332e9..d16f5e70891f9203ffb51c730a73a3ab571b43c2

I also noticed a couple of bugs which exist in `main` I fixed:
- Sometimes a `href` somewhere can be an empty string, which causes `isHrefExternal` to crash (probably while something is loading, I didn't look into it, this seems fine)
- A ProseMirror error when loading the Quick Notes page sometimes. I can't recreate this in the remote deployment but I can recreate it by refreshing in `main`

https://github.com/hashintel/hash/assets/37743469/525651cc-0b7f-4cbd-8ab5-18a0a2ea0962

Reply:
Forgot to say that I rewrote the reduce to something I find a bit easier to read since I had to update it for an archived filter anyway – `getEntityRevision` will return the latest edition from the subgraph if no timestamp is provided, which seems easier than comparing the timestamps manually.